device_samples/getting_started/src/connections/connection_string.js

```diff
@@ -0,0 +1,67 @@
+// Copyright (c) Microsoft. All rights reserved.
+// Licensed under the MIT license. See LICENSE file in the project root for full license information.
+
+'use strict';
+
+// Choose a protocol by uncommenting one of these transports.
+const Protocol = require('azure-iot-device-mqtt').Mqtt;
+// const Protocol = require('azure-iot-device-amqp').Amqp;
+// const Protocol = require('azure-iot-device-http').Http;
+// const Protocol = require('azure-iot-device-mqtt').MqttWs;
+// const Protocol = require('azure-iot-device-amqp').AmqpWs;
+
+const Client = require('azure-iot-device').Client;
+const Message = require('azure-iot-device').Message;
+
+// String containing Hostname, Device Id & Device Key in the following formats:
+//  "HostName=<iothub_host_name>;DeviceId=<device_id>;SharedAccessKey=<device_key>"
+const deviceConnectionString = process.env.IOTHUB_DEVICE_CONNECTION_STRING;
+
+// make sure we have a connection string before we can continue
+if (deviceConnectionString === '' || deviceConnectionString === undefined) {
+  console.error('\x1b[31m%s\x1b[0m', 'Missing device connection string');
```

Comment:
>'\x1b[31m%s\x1b[0m [](http://example.com/codeflow?start=16&length=18)

nit: I _love_ colored output for this kind of things, but shouldn't the `console.error` implementation be adding this?  You're assuming an ANSI console, and that's OK, but I've had to script these characters away before so I see this as a point of annoyance.